src/DefaultBuilder/src/WebHostEnvironment.cs

```diff
@@ -15,44 +16,47 @@ internal class WebHostEnvironment : IWebHostEnvironment
     {
         private static readonly NullFileProvider NullFileProvider = new();
 
+        private IFileProvider _contentRootFileProvider = NullFileProvider;
+        private IFileProvider _webRootFileProvider = NullFileProvider;
+        // ContentRootPath and WebRootPath are set to default! on
+        // initialization to match the behavior in HostingEnvironment.
+        private string _contentRootPath = default!;
+        private string _webRootPath = default!;
+
         public WebHostEnvironment(Assembly? callingAssembly)
         {
-            ContentRootPath = Directory.GetCurrentDirectory();
+            ContentRootPath = AppContext.BaseDirectory;
```

Comment:
@halter73 On closer examination of the code, it looks like the existing builder actually sets the `ContentRootPath` to `AppContext.BaseDirectory` by default.

https://github.com/dotnet/aspnetcore/blob/cb9ed02a31b7ea5c8ce8b0f54c5230d9a904739c/src/Hosting/Hosting/src/WebHostBuilder.cs#L266

`ResolveContentRootPath` will set the content root to `AppContext.BaseDirectory` if the user doesn't provide a path in the options.

Given this, I think it is sensible to set the root path to AppContext.BaseDirectory by default.

Reply:
I'm not sure the current builder just uses `AppContext.BaseDirectory`, but I haven't traced through all the logic. Here's what I see in a sample app though:

```C#
using System;
using Microsoft.AspNetCore.Hosting;
using Microsoft.Extensions.Hosting;

namespace WebApi
{
	public class Program
	{
		public static void Main(string[] args)
		{
            Console.WriteLine($"AppContext.BaseDirectory: {AppContext.BaseDirectory}");
			CreateHostBuilder(args).Build().Run();
		}

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
            .ConfigureWebHostDefaults(webBuilder =>
            {
                webBuilder.UseStartup<Startup>();
            });
    }
}
```

Output:

```
C:\dev\halter73\swagger\WebApi webroot ↑2  ~2 ❯ dotnet run
Building...
AppContext.BaseDirectory: C:\dev\halter73\swagger\WebApi\bin\Debug\net6.0\
...
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\dev\halter73\swagger\WebApi
```

Now if I set a relative path, it *does* seem to use `AppContext.BaseDirectory`:

If I add the following line to the `.ConfigureWebHostDefaults` callback

```C#
webBuilder.UseContentRoot("MyContentRoot");
```

Output:

```
C:\dev\halter73\swagger\WebApi webroot ↑2  ~2 ❯ dotnet run
Building...
AppContext.BaseDirectory: C:\dev\halter73\swagger\WebApi\bin\Debug\net6.0\
...
info: Microsoft.Hosting.Lifetime[0]
      Content root path: C:\dev\halter73\swagger\WebApi\bin\Debug\net6.0\MyContentRoot
```

This will fail with a `DirectoryNotFoundException` if `bin\Debug\net6.0\MyContentRoot` doesn't exist. I don't like this behavior and would like to make it more consistent for `WebApplicationBuilder` than it is for the generic host. I'm not positive what the generic host is using when you don't try to set a relative path for the content root.

Reply:
We're sticking to the current behavior per offline discussion.